### PR TITLE
Started ROL section.

### PR DESCRIPTION
**Pull request by dridzal**
First draft of the ROL section.  Needs a few additional bullets and references.  Added Trilinos members @dpkouri and @dridzal as co-authors.

### CODE REVIEW

**Comment by dridzal** [comment]
Also, @dpkouri should be added as the reviewer for this PR.  I can't assign reviewers for some reason.

**Review by srajama1 — APPROVED** [review]
Thanks @dridzal ! I can't add dpkouri as a reviewer.

**Comment by dridzal** [comment]
@srajama1 :  Can @dpkouri make edits and submit a PR?  We are missing a few paragraphs in the ROL section.

File: nonlinear_solvers.tex [hunk 1]
```diff
@@ -96,7 +96,76 @@ \subsection{Piro}
...
+
+The key design, performance and algorithmic features of ROL can be
+summarized as follows:
+
+\begin{itemize}
+\item
+\emph{Vector abstractions and matrix-free interface for universal applicability}.
+Similar to other Trilinos packages that implement abstract numerical
+algorithms, ROL's design is centered around an abstract linear algebra
+interface, through the ROL::Vector class, which enables the use of any
```
Comment (mayrmt) [inline nonlinear_solvers.tex:119]: Should we highlight source code by some special font? E.g. \texttt{ROL::Vector}?

File: nonlinear_solvers.tex [hunk 2]
```diff
@@ -96,7 +96,76 @@ \subsection{Piro}
...
+ROL~algorithm with any data type, such as the C++ std::vector, MPI-parallel
+data structures (e.g., Epetra, Tpetra, PETSc, HYPRE vectors), GPU-supporting data
+structures (e.g., Kokkos, ArrayFire), etc.  The abstract linear algebra
```
Comment (mayrmt) [inline nonlinear_solvers.tex:122]: ArrayFire has not been introduced in this paper so far, so one would need some context or citation.

**Review by mayrmt — COMMENTED** [review]
> Are the merges manual in this repo?

Yes, merging in this repository is manual.

File: nonlinear_solvers.tex (hunk 2 above)
Comment (dridzal) [reply nonlinear_solvers.tex:122]: @mayrmt : These are good suggestions.  Drew Kouri (@dpkouri) needs to contribute a few pieces to the ROL section, so is it ok if he makes your suggested changes?

File: nonlinear_solvers.tex (hunk 1 above)
Comment (dridzal) [reply nonlinear_solvers.tex:119]: Yes, we can do that in the next iteration.

**Pull request merged**